cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java

```diff
@@ -592,6 +592,18 @@ private void waitForJobAndHandleExceptionInternal(SQLEngineJob<?> job) throws SQ
     }
   }
 
+  /**
+   * Block until an SQL Engine job is completed.
+   *
+   * If an exception is thrown by this job, it is rethrown for the called to handle.
+   *
+   * @param job the job to wait for completion
+   * @throws SQLEngineException exception thrown when this job was stopped.
+   */
+  public void waitForJobAndThrowException(SQLEngineJob<?> job) throws SQLEngineException {
```

Comment:
If this method is only calling another method and nothing else, why do we need this new method?

Reply:
Fixed